tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/otextension/RotList.java

```diff
@@ -0,0 +1,118 @@
+package dk.alexandra.fresco.tools.ot.otextension;
+
+import dk.alexandra.fresco.framework.util.Drbg;
+import dk.alexandra.fresco.framework.util.Pair;
+import dk.alexandra.fresco.framework.util.StrictBitVector;
+import dk.alexandra.fresco.tools.ot.base.Ot;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A wrapper class for a list of random seed OTs. An instance of this class (once {@code send}
+ * and/or {@code receive} has been called) is meant to be used for several OT extension instances.
+ * <br/>
+ * Notice that an instance of this class is only meant to contain a single and static list of OTs.
+ * Thus a new instance must be made if one wishes to construct a list of new and random OTs.
+ */
+public class RotList {
+  private final int amount;
+  private final List<Pair<StrictBitVector, StrictBitVector>> sendMessages;
+  private final List<StrictBitVector> learnedMessages;
+  private final StrictBitVector choices;
+
+  private boolean sent = false;
+  private boolean received = false;
+
+  /**
+   * Prepares for doing a list of random OTs by internally constructing a list of
+   * random messages and random bit choices.
+   *
+   * @param rand
+   *          The randomness generator to use
+   * @param amount
+   *          The amount of OTs to construct
+   */
+  public RotList(Drbg rand, int amount) {
+    this.amount = amount;
+    sendMessages = new ArrayList<>(amount);
+    learnedMessages = new ArrayList<>(amount);
+    for (int i = 0; i < amount; i++) {
+      StrictBitVector seedZero = new StrictBitVector(amount, rand);
+      StrictBitVector seedOne = new StrictBitVector(amount, rand);
+      sendMessages.add(new Pair<>(seedZero, seedOne));
+    }
+    choices = new StrictBitVector(amount, rand);
+  }
+
+  /**
+   * Sends the prepared random OTs using {@code ot} as the underlying OT functionality.
+   *
+   * @param ot
+   *          The OT functionality to use for sending the random OTs
+   */
+  public void send(Ot ot) {
+    if (sent == true) {
+      throw new IllegalStateException("Seed OTs have already been sent.");
```

Comment:
Would be possible/reasonable to just hand the `ot` object to the constructor and do the sending/receiving at construction? This way there would be no reason to handle these exceptions.

Reply:
Maybe. The problem is that OT is asymmetric and thus one party must call send when the other calls receive and vice versa. Thus the constructor would also need the IDs from the OT functionality to do this properly. Furthermore this also requires the RotList to ALWAYS do OT in both directions. Something we might not want to do in general (MASCOT does it, but if we make 2PC protocols, we don't need it).

Reply:
Also I am not too keen on doing too much network interaction in constructors, both in regards to testing, but also in regards to usage.

Reply:
Ok, I guess you know better than I what problems it would cause :). However, at least then we should add to the documentation of these methods that they should only be called once/only called after send/recieve. Also, it would be nice to have getter methods for the sent/receive flags so you can test if the methods are safe to call.

Btw, we do network stuff in many constructors in, e.g., MASCOT, I do not see that as a problem.

Reply:
Added documentation in commit 0515efa5
I have not added gettter and setter methods since they are currently not needed and thus adding these would mean that we also had to test methods. In general I don't think it is a good idea to add dead code.
Regarding the network stuff: I personally prefer light constructors, in particular when we are at the primitive level (which I consider OT to be). The reason being that otherwise we must ensure that both parties are at the same place in the protocol at the same time, at the specific place where an object is constructed, _not_ when it is necessary used.

Reply:
Just the docs changed is fine by me. 

About the constructor, I think in this case this implementation does make sense to me. As you say, you may not always both send and receive. So you cant just do both at construction (unless you pass extra parameters to the constructor or something, which would just probably just be a mess anyway).